overlord/ifacestate/handlers.go

```diff
@@ -1120,8 +1122,10 @@ func (m *InterfaceManager) doAutoConnect(task *state.Task, _ *tomb.Tomb) error {
 	// if this is the case we can only proceed once the restart
 	// has happened or we may not have all the interfaces of the
 	// new core/base snap.
-	if err := snapstate.FinishRestart(task, snapsup); err != nil {
-		return err
+	if !m.preseed {
```

Comment:
wouldn't it be cleaner to have the check inside FinishRestart itself instead than here?

Reply:
Yes, but FinishRestart is a package-level func in snapstate, so it would need to use snapdenv.Preseeding() internally?

Reply:
I think that's fine. I think the consistency of having all decisions to skip restart related code in snapstate/boot is more important that avoiding snapdenv.Preseeding, unless you see some specific annoyance with the latter